### PR TITLE
DAOS-2592 doc: merge subsections of the deployment section

### PR DESCRIPTION
**Pull request by johannlombardi**
Restructure deployment section slightly to merge
the workflow with subsequent sub-sections.

Skip-build: true
Skip-test: true

Signed-off-by: Johann Lombardi <johann.lombardi@intel.com>

### CODE REVIEW

File: doc/admin/deployment.md [hunk 1]
```diff
@@ -43,14 +81,14 @@ while still utilizing the default locations is to create the
 required directories manually. To do this execute the following commands.
 
 daos_server:
-```
+```bash
 $ mkdir /var/run/daos_server
 $ chmod 0755 /var/run/daos_server
 $ chown user:user /var/run/daos_server (where user is the user you
     will run daos_server as)
 ```
```
Comment (jbrosenz) [inline doc/admin/deployment.md]: I have found that I need to also mkdir /tmp/daos_sockets, but I don't know if that's just me, or if that is a common step that everyone needs to do as well.

File: doc/admin/deployment.md [hunk 2]
```diff
@@ -87,282 +125,183 @@ in order to access NVMe SSDs through the SPDK framework.
...
+$ pdsh -w <server_list> daos_server start -o <config_file>`
+```
 
-### Step 5. Add device identifiers to server config file
+As for orterun, the list of storage nodes can be specified on the command line
+via the -H option. To start the DAOS server, run:
+```bash
+$ orterun --map-by node --mca btl tcp,self --mca oob tcp -np <num_servers>
+-H <server_list> --enable-recovery daos_server start -o <config_file>
+```
```
Comment (jbrosenz) [inline doc/admin/deployment.md]: Most recently, during some work to get ofi+verbs;ofi_rxm tests running, Alex had me also specify "--mca pml ob1" with orterun.  Can you check with Alex to see if that is a change that should be in there moving forward?

File: doc/admin/deployment.md (hunk 1 above)
Comment (johannlombardi) [reply doc/admin/deployment.md]: It depends on whether you have changed the default path for drpc/unix socket.

File: doc/admin/deployment.md (hunk 2 above)
Comment (johannlombardi) [reply doc/admin/deployment.md]: will check.

**Pull request merged**